Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -2988,14 +2967,37 @@ static Color parseColorFunction(CSSParserTokenRange& range, const CSSParserConte
     case CSSValueColorContrast:
         color = parseColorContrastFunctionParameters(colorRange, context);
         break;
-    case CSSValueColorMix:
-        color = parseColorMixFunctionParameters(colorRange, context);
-        break;
     default:
+        break;
+    }
+    if (!color.isValid())
         return { };
+
+    // Return a color.
+    range = colorRange;
+    return color;
     }
-    if (color.isValid())
-        range = colorRange;
+
+static std::optional<StyleColor> parseStyleColorFunction(CSSParserTokenRange& range, const CSSParserContext& context)
+{
+    CSSParserTokenRange colorRange = range;
+    CSSValueID functionId = range.peek().functionId();
+    std::optional<StyleColor> color;
+    switch (functionId) {
+    case CSSValueColorMix: {
+        auto optionalColor = parseColorMixFunctionParameters(colorRange, context);
+        if (optionalColor)
+            color = { *optionalColor };
+        break;
+    }
+    default:
+        break;
+    }
+    if (!color)
+        return { };
+
+    // Return a valid style color.
+    range = colorRange;
     return color;
 }
```

Comment:
Anything to do with color-mix should be done separately.

Reply:
Yes this was a leftover from the previous unified patch